jwst/wfss_contam/observations.py

```diff
@@ -10,10 +10,53 @@
 
 import logging
 
+from photutils.background import Background2D, MedianBackground
+from astropy.stats import SigmaClip
+
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
 
+def background_subtract(data, box_size=None, filter_size=(3,3), sigma=3.0, exclude_percentile=30.0):
+    """
+    Simple astropy background subtraction
+
+    Parameters
+    ----------
+    data : np.ndarray
+        2D array of pixel values
+    box_size : tuple
+        Size of box in pixels to use for background estimation. 
+        If not set, defaults to 1/5 of the image size.
+    filter_size : tuple
+        Size of filter to use for background estimation
+    sigma : float
+        Sigma threshold for background clipping
+    exclude_percentile : float
+        Percentage of masked pixels above which box is excluded from background estimation
+
+    Returns
+    -------
+    data : np.ndarray
+        2D array of pixel values with background subtracted
+
+    Notes
+    -----
+    Improper background subtraction in input _i2d image leads to extra flux
+    in the simulated dispersed image, and was one cause of flux scaling issues
+    in a previous version.
+    """
+    if box_size is None:
+        box_size = (int(data.shape[0]/5), int(data.shape[1]/5))
+    sigma_clip = SigmaClip(sigma=sigma)
+    bkg_estimator = MedianBackground()
+    bkg = Background2D(data, box_size, filter_size=filter_size,
```

Comment:
Is the `data` array used here the full i2d image array?  Just curious.

Reply:
yes, that's correct